### PR TITLE
zeromq-4.1.2, czmq-3.0.1

### PR DESCRIPTION
**Pull request by minrk**
latest stable release.

Also remove now-obsolete patch for 4.0.x.

### CODE REVIEW

**Comment by DomT4** [comment]
You're likely to need to add some `revision`s to the formulae that failed here.

**Comment by minrk** [comment]
Hm, it looks like zmq 4.1 breaks a lot of stuff.

**Comment by DomT4** [comment]
Not necessarily. Merge #40398 into this PR, then add `revision` bumps to:

* fontforge
* pushpin
* zurl

**Comment by nijikon** [comment]
@minrk you are working on this? I can pick this up.

**Comment by DomT4** [comment]
@nijikon Feel free. You'll have an easier time if you combine the `cmzq` and `zeromq` PRs into one.

**Comment by minrk** [comment]
@nijikon sorry, I dropped off as I found a number of bugs in zeromq-4.1.1 itself. This should be bumped to 4.1.2. I will try to get to it later today, but if you beat me to it, feel free.

**Comment by nijikon** [comment]
@minrk there is no point in us both working on it, so if you have time then do it, if not, I can start right away :)

**Comment by minrk** [comment]
Yup, I'll finish this one up now.

**Comment by nijikon** [comment]
@minrk awesome, thanks!

**Comment by minrk** [comment]
bumped zeromq to 4.1.2, merged czmq update into this one, and added revisions to fontforge, pushpin, and zurl.

**Comment by minrk** [comment]
I can't explain the build failures. At least one of them suggests that the library was compiled with libzmq-4.0, but libzmq-4.1 is installed (4.1 provides libzmq.5.dylib, 4.0 provides libzmq.4.dylib).

**Comment by DomT4** [comment]
The build was killed as it's quite a large one and the test-bot is currently backed up. It'll be re-queued within the next 24 hours.

You'll need to look at the `czmq` test failure - That one looks like the executable invoked may no longer exist or be installed automatically.

You can ignore the `fontforge` failure.

You need to bump the revision on the `m2sh` failure - That caused the `pushpin` failure.

**Comment by minrk** [comment]
Added revision to mongrel2, and made czmq run `make check` instead of `czmq_selftest`, which has been removed.

**Comment by DomT4** [comment]
Yosemite did not like the `make check` stage:

```
==> make
==> make check
 * zconfig: OK
 * zdigest: OK
 * zdir: make[2]: *** [check-local] Abort trap: 6
make[1]: *** [check-am] Error 2
make: *** [check-recursive] Error 1

```

File: Library/Formula/czmq.rb [hunk 1]
```diff
@@ -43,13 +42,9 @@ def install
 
     system "./autogen.sh" if build.head?
     system "./configure", *args
-    system "make", "install"
+    system "make"
+    system "make check"
```
Comment (DomT4) [inline Library/Formula/czmq.rb]: This sort of thing should be `system "make", "check"`. The comma has importance beyond just looking a little nicer.

Comment (minrk) [inline Library/Formula/czmq.rb]: Okay, I was following the pattern of other Formulae. I won't comment on whether it looks nicer or not.

**Comment by minrk** [comment]
I don't know what to say about the buildbot failing make check, other than my Yosemite machines pass just fine. I definitely regret including czmq in this PR. If it fails one more time, I'll remove it and start over with just libzmq.

**Comment by DomT4** [comment]
Merged in https://github.com/Homebrew/homebrew/commit/ac580d216e8fd721058d226bf32ae54c09e0e18f. Thanks for your contribution to Homebrew, we appreciate it! :smiley_cat:.

Sadly `zeromq` and `czmq` tend to need doing together because they cross-over in various formulae. Just for future reference the preferred checksum is `sha256`, Thanks!

**Comment by minrk** [comment]
Thanks, I'll update the next package I bump to use sha256.

**Pull request closed without merging**